### PR TITLE
feat(discover): Zerofill queries that are grouped by time

### PR DESCRIPTION
**Pull request by billyvg**
This will zerofill results when a query has the `rollup` parameter. This will give us more accurate charts when there are time periods without any events.

### CODE REVIEW

File: src/sentry/api/endpoints/organization_discover_query.py [hunk 1]
```diff
@@ -226,9 +226,10 @@ def handle_results(self, snuba_results, requested_query, projects):
                 ]
 
             for result in snuba_results['data']:
-                result['project.name'] = projects[result['project.id']]
-                if 'project.id' not in requested_query['selected_columns']:
-                    del result['project.id']
+                if 'project.id' in result:
```
Comment (billyvg) [inline src/sentry/api/endpoints/organization_discover_query.py]: It's now possible this doesn't not exist in the result for time buckets that do not have any data.

**Comment by lynnagara** [comment]
Also, would this mean that limits are kind of ignored on these types of queries?

**Comment by billyvg** [comment]
Can we bump the limit to 10k (that's what events is at)?

Limit is not ignored, you'll have a result object with `{time: <ts>}` and no other keys for every empty time bucket.

**Comment by lynnagara** [comment]
I don't have a strong opinion on the API limit, but can we keep the one that user can enter in the UI as 1000?

**Comment by lynnagara** [comment]
I don't think I fully understand `assumeNullAsZero` - do we still need that and how is it distinct from `assumeEmptyAsZero` - are these not being zerofilled?

**Comment by billyvg** [comment]
I removed the "assume as zero" option

**Comment by billyvg** [comment]
I'm not going to touch limit for now, widgets use `queryBuilder.fetch` (where limit is validated) - so we'll have to think about where to validate it.

**Review by lynnagara — APPROVED** [review]
LGTM

**Pull request merged**